banzai_floyds/extract.py

```diff
@@ -0,0 +1,267 @@
+from numpy.polynomial.legendre import Legendre
+from banzai_floyds.utils.fitting_utils import gauss, fwhm_to_sigma, Legendre2d
+
+
+def profile_gauss_fixed_width(params, x, sigma):
+    center, background_level = params
+    return gauss(x, center, sigma) + background_level
+
+
+def background_fixed_profile_center(params, x, center):
+    sigma, *coeffs = params
+    background = Legendre(coef=coeffs, domain=(np.min(x), np.max(x)))
+    return gauss(x, center, sigma) + background(x)
+
+
+def background_fixed_profile(params, coords, center, sigma, x_deg):
+    x, y = coords
+    x_coeffs = params[:x_deg + 1]
+    y_coeffs = np.append([1.0], params[x_deg + 1:])
+    background = Legendre2d(x_coeffs, y_coeffs, domains=((np.min(x), np.max(x)), (np.min(y), np.max(y))))
+    return gauss(y, center, sigma) + background(x, y)
+
+
+def background_only(params, coords, x_deg):
+    x, y = coords
+    x_coeffs = params[:x_deg + 1]
+    y_coeffs = np.append([1.0], params[x_deg + 1:])
+    background = Legendre2d(x_coeffs, y_coeffs, domains=((np.min(x), np.max(x)), (np.min(y), np.max(y))))
+    return background(x, y)
+
+
+def bins_to_bin_edges(bins):
+    bin_edges = bins['center'] - (bins['width'] / 2.0)
+    bin_edges = np.append(bin_edges, bins['center'][-1] + (bins['width'][-1] / 2.0))
+    return bin_edges
+
+
+def bin_data(data, uncertainty, wavelengths, orders, wavelength_bins):
+    binned_data = None
+    x2d, y2d = np.meshgrid(np.arange(data.shape[1]), np.arange(data.shape[0]))
+    for order_id, order_wavelengths in zip(orders.order_ids, wavelength_bins):
+        in_order = orders.data == order_id
+        # Throw away the data that is outside the first and last bins
+        min_wavelength = order_wavelengths[0]['center'] - (order_wavelengths[0]['width'] / 2.0)
+        max_wavelength = order_wavelengths[-1]['center'] + (order_wavelengths[-1]['width'] / 2.0)
+
+        in_order = np.logical_and(in_order, wavelengths.data > min_wavelength)
+        in_order = np.logical_and(in_order, wavelengths.data < max_wavelength)
+
+        y_order = y2d[in_order] - orders.center(x2d[in_order])[order_id - 1]
+        data_table = Table({'data': data[in_order], 'uncertainty': uncertainty[in_order],
+                            'wavelength': wavelengths.data[in_order], 'x': x2d[in_order],
+                            'y': y2d[in_order], 'y_order': y_order})
+        bin_number = np.digitize(data_table['wavelength'], bins_to_bin_edges(order_wavelengths))
+        data_table['wavelength_bin'] = order_wavelengths['center'][bin_number - 1]
+        data_table['wavelength_bin_width'] = order_wavelengths['width'][bin_number - 1]
+        data_table['order'] = order_id
+        if binned_data is None:
+            binned_data = data_table
+        else:
+            binned_data = vstack([binned_data, data_table])
+    return binned_data.group_by(('order', 'wavelength_bin'))
+
+
+def fit_profile(data, profile_width=4):
+    trace_centers = []
```

Comment:
I think this is redundant and unnecessary. 
Should be defined as an empty list by default in line 90 if there is no `order_data`.